src/target/target_internal.h

```diff
@@ -208,7 +208,15 @@ bool target_mem64_write16(target_s *target, target_addr64_t addr, uint16_t value
 bool target_mem64_write8(target_s *target, target_addr64_t addr, uint8_t value);
 bool target_check_error(target_s *target);
 
+#if defined(__MINGW32__) || defined(__MINGW64__) || defined(__CYGWIN__)
+#define TC_FORMAT_ATTR __attribute__((format(__MINGW_PRINTF_FORMAT, 2, 3)))
+#elif defined(__GNUC__) || defined(__clang__)
+#define TC_FORMAT_ATTR __attribute__((format(printf, 2, 3)))
```

Comment:
Thanks to @ALTracer for suggesting this macro.

Reply:
If you insist on mentioning contributors, then also shout out to @amyspark -- this is the implementation in PR1743 https://github.com/blackmagic-debug/blackmagic/commit/ce69efd0#diff-4caf95c6659dfe9a43bcd59e329b01aa444e4ae17669bddd063bf8d5b14bd319

Reply:
BTW, wouldn't it be better to move this macro and the one from @amyspark (located in gdb_packet.h) to some place where we can merge them in one and include it everywhere?

Reply:
Not really, the argument positions are different.

Reply:
But we'll have 2 `#define X_FORMAT_ATTR`  and we'll skip the duplication of `#if` / `#elif`. @dragonmux what do you think?

Reply:
They are in presently unrelated headers and while perhaps there's an argument to be made for a new header, this creates new inter-dependencies vs this approach and gives the headache that ALTracer mentions about having to say the printf parameter numbers everywhere these are used. Having them as separate macro defines in separate independent headers makes more sense in this context to us for now.